### PR TITLE
parachains_coretime: Expose `MaxXCMTransactWeight`

### PR DESCRIPTION
**Pull request by bkchr**
This should be configured on the runtime level and not somewhere inside the pallet.

### CODE REVIEW

File: polkadot/runtime/parachains/src/coretime/mod.rs [hunk 1]
```diff
@@ -110,6 +110,11 @@ pub mod pallet {
 		/// Something that provides the weight of this pallet.
 		type WeightInfo: WeightInfo;
 		type SendXcm: SendXcm;
+
+		/// Maximum weight for any XCM transact call that should be executed on the coretime chain.
+		///
+		/// Basically should be `max_weight(set_leases, reserve, notify_core_count)`.
+		type MaxXCMTransactWeight: Get<Weight>;
```
Comment (bkontur) [inline polkadot/runtime/parachains/src/coretime/mod.rs:117]: just a very very nit - to be aligned with the rest e.g. `MaxXcmEncodedSize`, `EnsureXcmOrigin`, `ExecuteXcm`, `SendXcm`...
```suggestion
		type MaxXcmTransactWeight: Get<Weight>;
```

**Review by acatangiu — APPROVED** [review]

File: polkadot/runtime/westend/src/lib.rs [hunk 2]
```diff
@@ -1188,6 +1188,7 @@ impl parachains_scheduler::Config for Runtime {
 
 parameter_types! {
 	pub const BrokerId: u32 = BROKER_ID;
+	pub MaxXCMTransactWeight: Weight = Weight::from_parts(200_000_000, 20_000);
```
Comment (seadanda) [inline polkadot/runtime/westend/src/lib.rs:1191]: Set lease has three reads and a write, independent of the ref time of the actual function on that runtime that already means a `ref_time` of `175_000_000`. Is it worth leaving a bit more of a buffer to account for fluctuations in benchmarking on top of this?

**Review by eskimor — APPROVED** [review]

File: polkadot/runtime/rococo/src/lib.rs [hunk 3]
```diff
@@ -1057,6 +1057,7 @@ impl parachains_scheduler::Config for Runtime {
 
 parameter_types! {
 	pub const BrokerId: u32 = BROKER_ID;
+	pub MaxXCMTransactWeight: Weight = Weight::from_parts(200_000_000, 20_000);
```
Comment (acatangiu) [inline polkadot/runtime/rococo/src/lib.rs:1060]:
```suggestion
	pub MaxXcmTransactWeight: Weight = Weight::from_parts(200_000_000, 20_000);
```

File: polkadot/runtime/rococo/src/lib.rs [hunk 4]
```diff
@@ -1066,6 +1067,7 @@ impl coretime::Config for Runtime {
 	type BrokerId = BrokerId;
 	type WeightInfo = weights::runtime_parachains_coretime::WeightInfo<Runtime>;
 	type SendXcm = crate::xcm_config::XcmRouter;
+	type MaxXCMTransactWeight = MaxXCMTransactWeight;
```
Comment (acatangiu) [inline polkadot/runtime/rococo/src/lib.rs:1070]:
```suggestion
	type MaxXcmTransactWeight = MaxXcmTransactWeight;
```

File: polkadot/runtime/westend/src/lib.rs (hunk 2 above)
Comment (acatangiu) [inline polkadot/runtime/westend/src/lib.rs:1191]:
```suggestion
	pub MaxXcmTransactWeight: Weight = Weight::from_parts(200_000_000, 20_000);
```

File: polkadot/runtime/westend/src/lib.rs [hunk 5]
```diff
@@ -1197,6 +1198,7 @@ impl coretime::Config for Runtime {
 	type BrokerId = BrokerId;
 	type WeightInfo = weights::runtime_parachains_coretime::WeightInfo<Runtime>;
 	type SendXcm = crate::xcm_config::XcmRouter;
+	type MaxXCMTransactWeight = MaxXCMTransactWeight;
```
Comment (acatangiu) [inline polkadot/runtime/westend/src/lib.rs:1201]:
```suggestion
	type MaxXcmTransactWeight = MaxXcmTransactWeight;
```

**Review by seadanda — APPROVED** [review]

**Comment by Polkadot-Forum** [comment]
This pull request has been mentioned on **Polkadot Forum**. There might be relevant details there:

https://forum.polkadot.network/t/kusama-v1-2-0-coretime-migration-xcm-failure-a-postmortem/7536/1

**Pull request merged**